### PR TITLE
refactor: Update field data output.

### PR DESCRIPTION
**Pull request by prmukherj**
Convert existing classes such as field_data.Vector and potentially BaseFieldData and its derivatives to use standard numpy arrays.

### CODE REVIEW

File: tests/test_field_data.py [hunk 1]
```diff
@@ -191,48 +191,51 @@ def test_field_data_objects_3d(new_solver_session) -> None:
...
+        data_types=[SurfaceDataType.FacesNormal], surfaces=[3, 5]
+    )
+    assert faces_normal_data[3].shape == (152, 3)
+    faces_normal_data[5].shape == (2001, 3)
 
     faces_connectivity_data = field_data.get_surface_data(
         data_types=[SurfaceDataType.FacesConnectivity], surfaces=["cold-inlet"]
     )
     assert faces_connectivity_data[5].node_count == 4
     assert (faces_connectivity_data[5].node_indices == [12, 13, 17, 16]).all()
```
Comment (prmukherj) [inline tests/test_field_data.py:220]: Just return the data and not the 'node_count' and 'node_indices'.

Comment (prmukherj) [reply tests/test_field_data.py]: Done.

File: tests/test_field_data.py [hunk 2]
```diff
@@ -191,48 +191,67 @@ def test_field_data_objects_3d(new_solver_session) -> None:
...
+    assert (
+        round(float(vertices_data[SurfaceDataType.Vertices]["cold-inlet"][5][0]), 2)
+        == -0.2
     )
-    assert round(float(faces_centroid_data[5].y), 2) == -0.18
 
-    faces_connectivity_data = field_data.get_surface_data(
-        data_types=[SurfaceDataType.FacesConnectivity], surfaces=["cold-inlet"]
+    vertices_and_faces_centroid_data = field_data.get_surface_data(
+        data_types=["vertices", "centroid"], surfaces=["hot-inlet", "cold-inlet"]
```
Comment (prmukherj) [inline tests/test_field_data.py:207]: The data type key should always return enums.

File: tests/test_field_data.py [hunk 3]
```diff
@@ -191,48 +191,67 @@ def test_field_data_objects_3d(new_solver_session) -> None:
         field_name="absolute-pressure", surfaces=["cold-inlet"]
     )
 
-    assert abs_press_data.size == 241
-    assert abs_press_data[120].scalar_data == 101325.0
+    assert abs_press_data["cold-inlet"].shape == (241,)
+    assert abs_press_data["cold-inlet"][120] == 101325.0
 
     vertices_data = field_data.get_surface_data(
         data_types=[SurfaceDataType.Vertices], surfaces=["cold-inlet"]
```
Comment (prmukherj) [inline tests/test_field_data.py:198]: Constraint the input to 'enum' only.

File: doc/source/user_guide/fields/field_data.rst [hunk 4]
```diff
@@ -99,50 +92,46 @@ You can call the ``get_scalar_field_data`` method to get scalar field data, such
...
 You can call the ``get_pathlines_field_data`` method to get pathlines field data.
 
 .. code-block:: python
 
-  >>> path_lines_data = field_data.get_pathlines_field_data(field_name="velocity", surface_name="cold-inlet")
-  >>> path_lines_data["vertices"].size
-  76152
-  >>> path_lines_data["lines"].size
+  >>> path_lines_data = field_data.get_pathlines_field_data(field_name="velocity", surfaces=["cold-inlet"])
+  >>> path_lines_data["cold-inlet"]["vertices"].shape
```
Comment (mkundu1) [inline doc/source/user_guide/fields/field_data.rst:124]: @prmukherj There is an inconsistency between the pathlines data and surface data return structure:

pathlines_data[<surface>][<vertices>]
surface_data[<vertices>][<surface>]

`<surface>` should preferably be the first index of the returning array for all APIs.

Comment (prmukherj) [reply doc/source/user_guide/fields/field_data.rst]: Done, Thank you.

**Review by mkundu1 — APPROVED** [review]

**Review by hpohekar — APPROVED** [review]

**Review by seanpearsonuk — APPROVED** [review]

**Pull request merged**